apps/dashboard/app/assets/javascripts/application.js

```diff
@@ -71,3 +72,63 @@ function installSettingHandlers(name) {
   var selector = 'input[type="range"][name="' + name + '"]';
   $(selector).change(storeSetting);
 }
+
+
+function promiseLoginToXDMoD(xdmodUrl){
+  return new Promise(function(resolve, reject){
+
+    var promise_to_receive_message_from_iframe = new Promise(function(resolve, reject){
+      window.addEventListener("message", function(event){
+        if (event.origin !== xdmodUrl){
+          console.log('Received message from untrusted origin, discarding');
+          return;
+        }
+        else if(event.data.application == 'xdmod'){
+          if(event.data.action == 'loginComplete'){
+            resolve();
+          }
+            else if(event.data.action == 'error'){
+              console.log('ERROR: ' + event.data.info);
+              //FIXME: what to pass to reject?
+              reject();
+          }
+        }
+      }, false);
+    });
+
+    fetch(xdmodUrl + '/rest/auth/idpredirect?returnTo=%2Fgui%2Fgeneral%2Flogin.php')
+      .then(response => response.ok ? Promise.resolve(response) : Promise.reject())
+      .then(response => response.json())
+      .then(function(data){
+        return new Promise(function(resolve, reject){
+          var xdmodLogin = document.createElement('iframe');
+          xdmodLogin.style = 'display:hidden;left: -10000';
```

Comment:
Sign into XDMoD and OnDemand, and you see the widgets. Then sign out of XDMoD, and reload the OnDemand dashboard page. The widgets hang in "LOADING..." state. And I can now see the iframe at the bottom of the page:

![screen 2020-07-01 at 4 36 44 PM](https://user-images.githubusercontent.com/512333/86289102-155d4780-bbb9-11ea-830f-4c6c7a027c84.png)

I had thought that in earlier tests this single sign-on trick worked at least in this state.